src/Monolog/Processor/WebProcessor.php

```diff
@@ -44,23 +44,21 @@ class WebProcessor
      */
     public function __construct($serverData = null, array $extraFields = null)
     {
-        if (null === $serverData) {
+        if ($serverData === null || count($serverData) === 0) {
```

Comment:
This is broken when counting on an ArrayAccess object. It is perfectly valid to implement ArrayAccess without implementing Countable, and you even count here without doing any validation whether it is allowed.